### PR TITLE
chore: Support exclude the label from payload

### PR DESCRIPTION
**Pull request by sarabala1979**

### CODE REVIEW

File: prometheus-pusher/main.go [hunk 1]
```diff
@@ -154,6 +157,10 @@ func (p *prometheusSink) createPusher(jobName string) (*push.Pusher, error) {
 	return pusher, nil
 }
 
+func parseStringToSlice(envValue string) []string {
+	return strings.Split(envValue, ",")
```
Comment (qhuai) [inline prometheus-pusher/main.go:161]: Will envValue be an empty string  in most cases? In that case, will the split() return an array of one element? Would return nil a better choice - performant later on?

File: prometheus-pusher/payload.go [hunk 2]
```diff
@@ -28,6 +28,19 @@ func (p *PrometheusPayload) mergeLabels(labels map[string]string) {
 	}
 }
 
+func (p *PrometheusPayload) excludeLabels(labels []string) {
+	if p.Labels == nil {
+		p.Labels = make(map[string]string)
```
Comment (qhuai) [inline prometheus-pusher/payload.go:33]: would a simple return work here?

File: prometheus-pusher/payload_test.go [hunk 3]
```diff
@@ -18,3 +18,38 @@ func TestConvertToPrometheusPayload(t *testing.T) {
...
 }
+
+func TestMergePrometheusLabelPayload(t *testing.T) {
+	JsonStr := `{"uuid":"35e0dc4603c845c9b999f5f669c64606","config_id":"test","composite_keys":["test_namespace","test_app","597b5bd8cc"],"timestamp":1701201827,"unified_anomaly":1.2,"data":{"namespace_app_rollouts_http_request_error_rate":null},"metadata":{"model_version":0,"artifact_versions":{"MinMaxScaler":"0","LSTMAE":"0","StdDevThreshold":"0"},"app":"test-app","intuit_alert":"true","namespace":"test-namespace","numalogic":"true","prometheus":"k8s-prometheus","rollouts_pod_template_hash":"597b5bd8cc"}}`
+	var origiObj OriginalPayload
+	err := json.Unmarshal([]byte(JsonStr), &origiObj)
+	assert.NoError(t, err)
+	prometheusPayload := origiObj.ConvertToPrometheusPayload("test")
+	labels := map[string]string{"label1": "value1", "label2": "value1"}
+	prometheusPayload.mergeLabels(labels)
```
Comment (qhuai) [inline prometheus-pusher/payload_test.go:29]: consider to add prometheusPayload.excludeLabels(nil) here to provide extra coverage.

**Review by qhuai — APPROVED** [review]
Added some minor comments. LGTM overall.

File: prometheus-pusher/main.go (hunk 1 above)
Comment (sarabala1979) [reply prometheus-pusher/main.go:164]: It returns an empty array len=0. we are good. but add check

File: prometheus-pusher/payload.go (hunk 2 above)
Comment (sarabala1979) [reply prometheus-pusher/payload.go]: good catch

File: prometheus-pusher/payload_test.go (hunk 3 above)
Comment (sarabala1979) [reply prometheus-pusher/payload_test.go:30]: done

**Pull request merged**